### PR TITLE
Add notes

### PR DESCRIPTION
**Pull request by Notdeath**
## What have you changed? (mandatory)

```rust
--  Added some comments 
-- 
```

## What are the type of the changes? (mandatory)
no
## How has this PR been tested? (mandatory)
don`t need
## Does this PR affect documentation (docs) update? (mandatory)
no

## Does this PR affect tidb-ansible update? (mandatory)
no

### CODE REVIEW

**Comment by ice1000** [comment]
Please sign-off all your commits

File: src/storage/mvcc/reader/mod.rs [hunk 1]
```diff
@@ -84,7 +84,9 @@ impl<S: Snapshot> MvccReader<S> {
         self.key_only = key_only;
     }
 
+    /// Find data based on key and submission time ts
     pub fn load_data(&mut self, key: &Key, ts: u64) -> Result<Value> {
+        // usually storage index
```
Comment (ice1000) [inline src/storage/mvcc/reader/mod.rs]: What does this mean?

File: src/storage/mvcc/reader/mod.rs [hunk 2]
```diff
@@ -111,6 +113,7 @@ impl<S: Snapshot> MvccReader<S> {
         Ok(res)
     }
 
+    /// find the lock by Key
```
Comment (ice1000) [inline src/storage/mvcc/reader/mod.rs]: Capitalize?

File: src/storage/mvcc/reader/mod.rs [hunk 3]
```diff
@@ -236,6 +241,7 @@ impl<S: Snapshot> MvccReader<S> {
         }
         loop {
             match self.seek_write(key, ts)? {
+                // find commit data
```
Comment (ice1000) [inline src/storage/mvcc/reader/mod.rs]: Capitalize?

File: src/storage/mod.rs [hunk 4]
```diff
@@ -663,6 +664,7 @@ impl<E: Engine> Storage<E> {
                         ctx.get_isolation_level(),
                         !ctx.get_not_fill_cache(),
                     );
+                    //the get methoe in storage/mvcc/reader/mod.rs, get data from snapshot
```
Comment (ice1000) [inline src/storage/mod.rs]: Spacing?

**Comment by ice1000** [comment]
BTW, what's the meaning of your issue title?

File: src/storage/mvcc/reader/mod.rs (hunk 1 above)
Comment (Notdeath) [reply src/storage/mvcc/reader/mod.rs]: if key_only is true,there is a great possibility that the index is stored, so it returns Ok(vec![]);

**Comment by ngaut** [comment]
Please change the title of the PR.

**Comment by ice1000** [comment]
>           Please sign-off all your commits

Please do this.

File: src/storage/mod.rs [hunk 5]
```diff
@@ -663,6 +664,7 @@ impl<E: Engine> Storage<E> {
                         ctx.get_isolation_level(),
                         !ctx.get_not_fill_cache(),
                     );
+                    // the get methoe in storage/mvcc/reader/mod.rs
```
Comment (ice1000) [inline src/storage/mod.rs]: Capitalize

File: src/storage/mod.rs [hunk 6]
```diff
@@ -829,6 +831,7 @@ impl<E: Engine> Storage<E> {
         Ok(())
     }
 
+    /// Put、delete、lock port,
```
Comment (ice1000) [inline src/storage/mod.rs]: Can you use pure ASCII?

File: src/storage/mvcc/reader/mod.rs [hunk 7]
```diff
@@ -84,7 +84,9 @@ impl<S: Snapshot> MvccReader<S> {
         self.key_only = key_only;
     }
 
+    /// Find data based on key and submission time ts
     pub fn load_data(&mut self, key: &Key, ts: u64) -> Result<Value> {
+        // if key_only is true,there is a great possibility that the index is stored
```
Comment (ice1000) [inline src/storage/mvcc/reader/mod.rs]: Capitalize

File: src/storage/mvcc/reader/mod.rs [hunk 8]
```diff
@@ -255,7 +260,7 @@ impl<S: Snapshot> MvccReader<S> {
             }
         }
     }
-
+
```
Comment (ice1000) [inline src/storage/mvcc/reader/mod.rs]: Please don't make changes like this.

File: src/storage/txn/scheduler.rs [hunk 9]
```diff
@@ -302,7 +307,7 @@ impl<E: Engine> Scheduler<E> {
     }
 
     fn on_receive_new_cmd(&mut self, cmd: Command, callback: StorageCb) {
-        // write flow control
+        // write flow control
```
Comment (ice1000) [inline src/storage/txn/scheduler.rs]: ditto

File: src/storage/txn/scheduler.rs [hunk 10]
```diff
@@ -476,6 +481,9 @@ impl<E: Engine> Runnable<Msg> for Scheduler<E> {
     }
 }
 
+/// Generate a Lock for each task.
+/// First, find all the keys involved in the task
+/// Then,the gen_lock method maps all key hashes to latches and sorts them in ascending order
```
Comment (ice1000) [inline src/storage/txn/scheduler.rs]: Space after `,`

File: src/storage/mvcc/reader/mod.rs [hunk 11]
```diff
@@ -84,7 +84,9 @@ impl<S: Snapshot> MvccReader<S> {
         self.key_only = key_only;
     }
 
+    /// Find data based on key and submission time ts
     pub fn load_data(&mut self, key: &Key, ts: u64) -> Result<Value> {
+        // If key_only is true,there is a great possibility that the index is stored
```
Comment (ice1000) [inline src/storage/mvcc/reader/mod.rs]: Spacing

File: src/storage/txn/scheduler.rs [hunk 12]
```diff
@@ -206,12 +206,16 @@ impl<E: Engine> Scheduler<E> {
...
+    /// The task gets all the resources it needs, 
+    /// and it goes out of the queue for execution.
     fn dequeue_task(&mut self, cid: u64) -> Task {
         let task = self.pending_tasks.remove(&cid).unwrap();
         assert_eq!(task.cid, cid);
         task
     }
 
+    /// The task enters the scheduling wait queue 
+    /// Waiting to get related latches
```
Comment (ice1000) [inline src/storage/txn/scheduler.rs]: Maybe `,` or `.` are missing?

Comment (Notdeath) [reply src/storage/txn/scheduler.rs]: let me try

Comment (Notdeath) [reply src/storage/txn/scheduler.rs]: thanks

**Comment by zhangjinpeng1987** [comment]
Please run `make dev` at your computer first.

File: src/storage/mvcc/reader/mod.rs (hunk 8 above)
Comment (ice1000) [reply src/storage/mvcc/reader/mod.rs]: Ping @Notdeath

File: src/storage/txn/scheduler.rs (hunk 9 above)
Comment (ice1000) [reply src/storage/txn/scheduler.rs]: Ping @Notdeath

File: src/storage/mod.rs [hunk 13]
```diff
@@ -828,7 +830,7 @@ impl<E: Engine> Storage<E> {
         self.schedule(cmd, StorageCb::Boolean(callback))?;
         Ok(())
     }
-
+
```
Comment (ice1000) [inline src/storage/mod.rs]: Please remove trailing spaces.

File: src/storage/mod.rs [hunk 14]
```diff
@@ -866,6 +869,13 @@ impl<E: Engine> Storage<E> {
         commit_ts: u64,
         callback: Callback<()>,
     ) -> Result<()> {
+        if commit_ts <= lock_ts {
+            callback(Err(Error::InvalidTxnTso {
+                    start_ts: lock_ts,
+                    commit_ts,
+                }));
+            return Ok(());
+        }
```
Comment (ice1000) [inline src/storage/mod.rs]: This is not a comment. You should mention this change in your PR.

File: src/storage/mod.rs [hunk 15]
```diff
@@ -663,6 +664,7 @@ impl<E: Engine> Storage<E> {
                         ctx.get_isolation_level(),
                         !ctx.get_not_fill_cache(),
                     );
+                    // The get methoe in storage/mvcc/reader/mod
```
Comment (siddontang) [inline src/storage/mod.rs]: methoe?

Comment (Notdeath) [reply src/storage/mod.rs]: Clever

**Comment by huachaohuang** [comment]
Seems this PR is not active, let's close it.

**Pull request closed without merging**